src/App.tsx

```diff
@@ -8,37 +8,159 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment, CommentData } from './types/Comment';
+import { client } from './utils/fetchClient';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [currentPost, setCurrentPost] = useState<Post | null>(null);
+  const [loadingPosts, setLoadingPosts] = useState(false);
+  const [postsError, setPostsError] = useState('');
+  const [sideBarOpened, setSideBarOpened] = useState(false);
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [loadingComments, setLoadingComments] = useState(false);
+  const [commentsError, setCommentsError] = useState('');
+  const [newCommentForm, setNewCommentForm] = useState(false);
+  const [loadingComment, setLoadingComment] = useState(false);
+
+  useEffect(() => {
+    async function fetchUsers() {
+      const data = await client.get<User[]>('/users');
+
+      setUsers(data);
+    }
+
+    fetchUsers();
+  }, []);
+
+  const loadPosts = async (user: User) => {
+    if (user === selectedUser) {
+      return;
+    }
+
+    try {
+      setSideBarOpened(false);
+      setLoadingPosts(true);
+      setPostsError('');
+
+      const data = await client.get<Post[]>(`/posts?userId=${user.id}`);
+
+      setPosts(data);
+    } catch (error) {
+      setPosts([]);
+      setPostsError('Failed to load posts');
+    } finally {
+      setLoadingPosts(false);
+    }
+  };
+
+  const loadComments = async (post: Post) => {
+    try {
+      setLoadingComments(true);
+      setCommentsError('');
+
+      const data = await client.get<Comment[]>(`/comments?postId=${post.id}`);
+
+      setComments(data);
+      setCurrentPost(post);
+    } catch (error) {
+      setComments([]);
+      setCommentsError('Failed to load comments');
+    } finally {
+      setLoadingComments(false);
+    }
+  };
+
+  const addComment = async ({
+    postId, name, email, body,
+  }: CommentData) => {
+    try {
+      setLoadingComment(true);
+      const response: Comment = await client.post(
+        '/comments', {
+          postId, name, email, body,
+        },
+      );
+
+      setComments((prevComments) => {
+        if (prevComments !== null) {
+          return [...prevComments, response];
+        }
+
+        return [response];
+      });
+    } catch (error: any) {
```

Comment:
Don't use any for types. Add the correct type here 
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/70009205/3b5d4c7f-f23c-43b3-af98-c76bfb7a3f5b)